openmdao/components/exec_comp.py

```diff
@@ -461,8 +465,9 @@ def compute_partials(self, inputs, partials):
                 self._residuals.set_val(0.0)
                 self.compute(pwrap, uwrap)
 
-                for u in out_names:
-                    partials[(u, input)] = imag(uwrap[u] * inv_stepsize).flat
+                for out, inp in self._declared_partials.keys():
```

Comment:
It seems more efficient to keep the `for u in out_names` loop and just add a check for `if (u, input) in self._declared_partials: ...` rather than looping over all declared partials for each input and updating partials when the inputs match.

Reply:
Good idea. Changed.